runtime-common/search-index.ts

```diff
@@ -328,6 +332,28 @@ export class SearchIndex {
       readdir: this.readdir,
       readFileAsText: this.readFileAsText,
     });
+    this.#startedPhase1.fulfill();
+    let numUnloadedLinks = this.findCardsWithUnloadedLinks().length;
+    while (numUnloadedLinks > 0) {
+      this.#currentRun = await CurrentRun.incrementalUnloadedLinks(
+        this.#currentRun
+      );
+      let nextUnloadedLinks = this.findCardsWithUnloadedLinks().length;
+      if (nextUnloadedLinks >= numUnloadedLinks) {
+        break;
+      }
+      numUnloadedLinks = nextUnloadedLinks;
+    }
+    this.#startedPhase2.fulfill();
```

Comment:
this is the key logic where we implement the 2 phase indexing for a fresh search index. The idea is to write error docs for all the instances with links on the first pass, and then in the 2nd phase we continually incrementally index the instances with unloaded links, making sure that after each pass we have less errors for missing links than in the previous pass. we stop when we have no more unloaded links, or we are unable to whittle down the unloaded links any farther.